### PR TITLE
Enhance user guide

### PR DESCRIPTION
**Pull request by alejandrox1**
/kind documentation

Signed-off-by: Jorge Alarcon Ochoa <alarcj137@gmail.com>

### CODE REVIEW

File: docs/user/README.md [hunk 1]
```diff
@@ -18,22 +18,69 @@ By default, the cluster will be given the name `kind-1`. "1," here, is the
 default context name. 
 Use the `--name` flag to assign the cluster a different context name.
 
-Once a cluster is created you can use [kubectl][kubectl] to interact with it by
-using the configuration file generated by `kind`:
+`kind` also allows you to customize your Kubernetes cluster by using a kind
```
Comment (neolit123) [inline docs/user/README.md]: `using a kind ` -> `using a`

File: docs/user/README.md [hunk 2]
```diff
@@ -18,22 +18,69 @@ By default, the cluster will be given the name `kind-1`. "1," here, is the
...
-using the configuration file generated by `kind`:
+`kind` also allows you to customize your Kubernetes cluster by using a kind 
+configuration file. 
+For a sample kind configuration file see 
+[kind-example-config][kind-example-config].
+To use a kind configuration file use the `--config` flag and pass the path to   
+the file.
+
+
+
```
Comment (neolit123) [inline docs/user/README.md]: can omit 2 of these new lines.

File: docs/user/README.md [hunk 3]
```diff
@@ -18,22 +18,69 @@ By default, the cluster will be given the name `kind-1`. "1," here, is the
...
 export KUBECONFIG="$(kind get kubeconfig-path)"
 kubectl cluster-info
 ```
 
 `kind get kubeconfig-path` returns the location of the generated confguration
 file.
+If you gave a non-default cluster context name to your cluster, then you can
+specify the name by using the `--name` flag.
+
+To see all the clusters yoy have created, you can use the `get clusters`
```
Comment (neolit123) [inline docs/user/README.md]: typo `yoy`

File: docs/user/README.md [hunk 4]
```diff
@@ -18,22 +18,69 @@ By default, the cluster will be given the name `kind-1`. "1," here, is the
...
+command.
+
+For example, let's say you create two clusters:
+```
+$ kind create cluster        # Default cluster context name is "1".
+...
+
```
Comment (neolit123) [inline docs/user/README.md]: omit empty line.

File: docs/user/README.md [hunk 5]
```diff
@@ -18,22 +18,69 @@ By default, the cluster will be given the name `kind-1`. "1," here, is the
...
 file.
+If you gave a non-default cluster context name to your cluster, then you can
+specify the name by using the `--name` flag.
+
+To see all the clusters yoy have created, you can use the `get clusters`
+command.
+
+For example, let's say you create two clusters:
+```
+$ kind create cluster        # Default cluster context name is "1".
```
Comment (neolit123) [inline docs/user/README.md]: leave a single space after `cluster` and before `#`?

File: docs/user/README.md [hunk 6]
```diff
@@ -63,8 +110,12 @@ the resulting node image using the flag `--image`.
 If you previously changed the name and tag of the base image, you can use here
 the flag `--base-image` to specify the name and tag you used.
 
+
+
+
```
Comment (neolit123) [inline docs/user/README.md]: no need for these empty lines?

File: docs/user/README.md [hunk 7]
```diff
@@ -85,6 +136,42 @@ $ kind build base-image --image base:v0.1.0
...
+### Exporting `kind`'s Logs
+`kind` has the ability to export all `kind` related logs for you to explore.
+To export all logs from the default cluster (context name "1"):
+```
+$ kind export logs
+Exported logs to: /tmp/396758314
+```
+
+Like all other commands, if you want to perform the action on a cluster with a
+different context name then, use the `--name` flag.
```
Comment (neolit123) [inline docs/user/README.md]: `then, use` -> `use`

File: docs/user/README.md [hunk 8]
```diff
@@ -85,6 +136,42 @@ $ kind build base-image --image base:v0.1.0
...
+
+As you can see, `kind` placed all the logs for the cluster `kind-1` in a
+temporary directoy. If you want to specify a location then simply add the path
```
Comment (neolit123) [inline docs/user/README.md]: typo `directoy`

File: docs/user/README.md [hunk 9]
```diff
@@ -85,6 +136,42 @@ $ kind build base-image --image base:v0.1.0
...
+    ├── containers
+    ├── docker.log
+    ├── inspect.json
+    ├── journal.log
+    ├── kubelet.log
+    ├── kubernetes-version.txt
+    └── pods/
+```
+The logs contain information about the Docker host, the containers running 
+`kind`, the kubernetes cluster itself, etc.
```
Comment (neolit123) [inline docs/user/README.md]: `kubernetes` should be uppercase.

File: docs/user/README.md [hunk 10]
```diff
@@ -85,6 +136,42 @@ $ kind build base-image --image base:v0.1.0
...
+
+
```
Comment (neolit123) [inline docs/user/README.md]: one empty line can be omitted.

**Comment by alejandrox1** [comment]
Thank you @neolit123

**Comment by BenTheElder** [comment]
/ok-to-test

apologies, I've been chasing after upstream breakages (all fixed now!) and will be out most of tomorrow. thanks for continuing to work on this, I will get back around to it soon! 😬

File: docs/user/README.md [hunk 11]
```diff
@@ -18,22 +18,66 @@ By default, the cluster will be given the name `kind-1`. "1," here, is the
 default context name. 
 Use the `--name` flag to assign the cluster a different context name.
 
-Once a cluster is created you can use [kubectl][kubectl] to interact with it by
-using the configuration file generated by `kind`:
+`kind` also allows you to customize your Kubernetes cluster by using a
```
Comment (BenTheElder) [inline docs/user/README.md]: this should probably be in the advanced docs for now, nothing in here should be touched by a typical user. I want to make it possible to get started with `kind` with the minimum of information, with details covering complex usage saved for later in the docs.

Comment (alejandrox1) [reply docs/user/README.md]: Moved that to its own section. What do you think?

**Comment by BenTheElder** [comment]
needs a rebase to merge it looks like

**Pull request merged**